### DISCUSSION TITLE
ReadMe Template Discussion

### DISCUSSION THREAD

**Discussion by sarabond** [pds-data-dictionaries/PDS-Data-Dictionaries.github.io#12] [Ideas]
We (Team 4, Kiri, Mitch, and Sara) would like to note that Jordan did a tremendous job getting the new ReadMe template out in real time, but as we review it now, we see some room for improvement. 

1. The revised ReadMe template points to src/readme.md where it expects we'll have the information for the steward. We need to make that clear when you revise the main readme you have to update the src/readme.md. 

2. The current ReadMe template has a link to the pds.nasa.gov site, specific for the dictionary/namespace. The link includes the IM version. Currently the readme needs to be updated for each version. For example : https://pds.nasa.gov/datastandards/dictionaries/index-1.17.0.0.shtml#survey

3.  In the new template under Documentation the bullets are: 
  -  User Guide
  - Detailed Documentation
  - Other PDS Namespace Documentation
 
 (a) We believe the User Guide is the detailed documentation. We recommend the following change:

  - Namespace Outline
  - User Guide -- Detailed Documentation
  - Other PDS Namespace Documentation
 
 (b) The third bullet currently points to https://pds-data-dictionaries.github.io/ . We think this is inappropriate. We're not sure what we want to link to for "Other PDS Namespace Documentation," but it might be https://pds.nasa.gov/datastandards/dictionaries/index-1.17.0.0.shtml
Something more informative than "Other...." would be useful. 

4. Places in the guide that point to external links are not specified. Essentially we don't tell users they are leaving the site, but we should provide a warning that it is an external link. 

5.  Latest release points to pds.nasa.gov and to Github, and it's not clear why, or to which one a user should go. On Github it points to zip files or tar.gz, but not to the build directory. Do we need a link to the build directory as well? What do we mean by "latest" release? For example, the Github Geometry/latest points to the 1930 zip but there is a later release, 1950, available in build/development, and also at pds.nasa.gov. The point is there is confusion over what we want to communicate here. We need better defined words in the ReadMe file.

**Comment by sarabond** [comment]
We already incorporated some of our suggestions above to the Survey Discipline Namespace README file at https://github.com/pds-data-dictionaries/ldd-survey/blob/dmsp-updates/README.md

**Reply by jordanpadams** [reply]
@sarabond this is great! Once folks have some ideas of things they want to change, feel free to create a PR in the ldd-template repo. Happy to update however folks deem it to be useful.

**Reply by sarabond** [reply]
Awesome, will do! PR is short for pull request, just to be sure, right?

**Reply by jordanpadams** [reply]
@sarabond correct

**Comment by acraugh** [comment]
@sarabond , the DMSP group had very different ideas about "user guide" vs. "detailed documentation". The user guide was described as mid-level, how-to-use-it type documentation that would have a more tutorial approach and view the namespace in terms of the functionality it provides and how to apply it in designing labels. The "detailed documentation" would describe the namespace itself in intimate detail, attribute by attribute, with all constraints and valid usages completely described, along with some discussion of development philosophy and assumptions made in how the namespace relates to the greater Information Model. I would liken it to a "programmer's manual". The primary users for the detailed documentation would be software developers, stewards of the namespace (since stewardship is passed on as time goes by), PDS nodes developing customized search engines, and the configuration control board for the namespace. The two documents would provide different levels of information for very different target audiences.

**Comment by mit3ch** [comment]
Hi Anne. I'm the one who thought the User Guide contained the details. For some reason, I thought the mid-level document was referred to as the namespace outline (or overview), and it's primary function was to help a user decide which portion of the namespace was appropriate for the specific task. We all agree on what roughly the mid-level and in-the-weeds documentation contain, but apparently I got the labels wrong.

Mitch

Dr. Mitch Gordon
SETI Institute
Deputy Manager
PDS Ring-Moon Systems Node
276-393-8822
Pronouns: he, him, his

**Comment by wkiri** [comment]
By the way, we also noticed that referring to a user (e.g., the namespace steward) in the typical github way (@username), as recommended in the new README template, does not generate a clickable link in the README.md, unfortunately.  An alternate solution is to explicitly include a link to https://github.com/username (e.g., https://github.com/wkiri).  I suggest this as an update to the README template (unless someone has a better idea).  Thanks!

**Reply by acraugh** [reply]
Gonna try a two-birds approach:

@mit3ch, I think this merely re-emphasizes the need to generate some post-workshop documentation to put stuff in public writing, so add it to the list. Or start a list, if it doesn't already exist...

@wkiri, or probably this is a question for @jordanpadams, where would/should that clickable link on a handle take you? Are there options? Seems like if it's going to be clickable at that level of visibility, it has to do something that will work for people without github handles who click it.

**Reply by wkiri** [reply]
> @wkiri, or probably this is a question for @jordanpadams, where would/should that clickable link on a handle take you? Are there options? Seems like if it's going to be clickable at that level of visibility, it has to do something that will work for people without github handles who click it.

@acraugh Do we have any Namespace stewards who do not have github accounts?

**Reply by wkiri** [reply]
Note, you can reply to individual comments and it threads the discussion, nice!

**Comment by acraugh** [comment]
Not by the time Jordan's done, @wkiri, but my point is that if the user clicking doesn't also have a github handle, what would they expect to happen? I'm not sure how restricted communications might be for people without github logins. They can certainly find the readme.md without having a login, but the communication might be limited to one-way, which is not usually all that satisfying if you're hoping for follow-up. I suppose I could try it and see what happens (mwah-hah-hah!)...

**Reply by wkiri** [reply]
The user who is clicking doesn't need a github login.  It goes to the user-being-clicked's github page, which is visible to all.

**Comment by mit3ch** [comment]
Do we want to provide the steward's email in the readme instead of the handle? In addition to?

Mitch

Dr. Mitch Gordon
SETI Institute
Deputy Manager
PDS Ring-Moon Systems Node
276-393-8822
Pronouns: he, him, his

**Reply by wkiri** [reply]
> Do we want to provide the steward's email in the readme instead of the handle? In addition to?

If the goal is direct communication, then email is the better solution here for sure!  (The @username usage just tells you who the person is; it doesn't let you send a message either.)

**Reply by jordanpadams** [reply]
👍 agreed. probably email address / contact info is a better route for the readme

**Comment by beatricemueller** [comment]
Now that @acraugh has a picture (I reserve my judgement) we need to get @mit3ch to also get on board. Let's see if he can top Anne.

**Comment by acraugh** [comment]
@jordanpadams, @sarabond, @wkiri, and @mit3ch - I created a pull request on the ldd-template repo to tweak the repo README.MD template. Have a look if you have the time. I did not attempt to address the email/handle issue - haven't had time to test it out. Thanks!

**Reply by wkiri** [reply]
Thanks, @acraugh .  The email/handle issue can be as simple as writing

> Steward: a@b.c

which Github converts to a clickable mailto link.

**Comment by wkiri** [comment]
I noticed a small error in the README template.  It contains this line:
```
* [GitHub](../../../releases/latest)
```
which should be
```
* [GitHub](../../releases/latest)
```
I don't see how to open an issue for the template repo, and don't have time right now to create a fix and PR, so I thought I would capture it here in case someone else can address it.  Thanks!

**Comment by acraugh** [comment]
I see it, @wkiri. I'll take care of it.

**Comment by wkiri** [comment]
Thanks so much, @acraugh !
